### DISCUSSION TITLE
Response status

### DISCUSSION THREAD

**Discussion by AmnesiaZero** [def-studio/telegraph#543] [Q&A]
Is there is a way to set json response status to message()->send() method? Some messages by bot had no status and it's casuing resending same message over and over again

![image](https://github.com/defstudio/telegraph/assets/90193990/8fcefce3-8d2c-4ec6-96d7-31172afbb70e)

**Comment by fabio-ivona** [comment]
Hi, could you tell me which kind of messages have no status?

telegraph should send a 204 status on all requests

**Reply by AmnesiaZero** [reply]
This message also has a long response time. After user sends message,bot sends response in about 1.5 minutes

**Comment by AmnesiaZero** [comment]
Messages with files attached to them and without text

**Comment by AmnesiaZero** [comment]
![image](https://github.com/defstudio/telegraph/assets/90193990/36bfdd7c-c6db-4711-a7f6-e3682e47f425)

**Comment by AmnesiaZero** [comment]
Whole method:
        public function uploadDocument(): void
    {
        set_time_limit(0);
        $chatId = $this->getChatId();
        $messageDocument = $this->message->document();
        Log::debug('data = '.$this->logArray($this->data));
        $documentName = $messageDocument->filename();
        Log::debug($this->logArray($this->message));
        $documentFolderPath = storage_path('app/public');
        $documentPath = $documentFolderPath.'/'.$documentName;
        Log::debug('document path = '.$documentPath);
        $this->telegraph->store($messageDocument, $documentFolderPath, $documentName);
        $response = $this->telegraph->chat($chatId)->message('Загрузка......')->send();
        $messageId = $response->telegraphMessageId();
        Log::debug('download status = '.$response->status());
        $client = $this->getVkrClient();
        $document = new Document($client,true);
        if(!file_exists($documentPath)){
            $this->errorMessage('Файл не найден',$chatId,$messageId);
            return;
        }
        if(!$document->uploadDocument($documentPath,['organization_id' => 0])){
            $this->errorMessage($document->getMessage(),$chatId,$messageId);
            return;
        }
        Log::debug('test commit');
        Log::debug('Отправил документ');
        $report = new Report($client,true);
        $documentId = $document->getId();
        if (!$report->get($documentId)) {
            $this->errorMessage($report->getMessage(), $chatId, $messageId);
            return;
        }
        Log::debug("Отправка отчёта");
        $response =  $this->telegraph->chat($chatId)->deleteMessage($messageId)->send();
        Log::debug('delete status = '.$response->status());
        $showDomain = config('show_api.domain');
        $response = $this->telegraph->chat($chatId)->message("Получен отчёт:\n" . $report->toString() . "\n\nПолноценный отчёт можете посмотреть по ссылке - " . $showDomain . "/report/".$documentId)->keyboard(Keyboard::make()->buttons([
            Button::make('В начало')->action('start')->param('chat_id', $chatId),
            Button::make('Отправить другую работу')->action('vkrsmart')->param('chat_id', $chatId)
        ]))->send();
        Log::debug('final status = '.$response->status());
    }
